api/trading_data_v2.go

```diff
@@ -193,6 +195,143 @@ func entityMarketDataListToProtoList(list []entities.MarketData) (*v2.MarketData
 	return &connection, nil
 }
 
+func (t *tradingDataServiceV2) ObserveMarketsDepth(req *v2.ObserveMarketsDepthRequest, srv v2.TradingDataService_ObserveMarketsDepthServer) error {
+	// Wrap context from the request into cancellable. We can close internal chan on error.
+	ctx, cancel := context.WithCancel(srv.Context())
+	defer cancel()
+
+	for _, marketId := range req.MarketIds {
+		_, err := validateMarketSQL(ctx, marketId, t.marketService)
```

Comment:
Just looked at validateMarketSQL, it seems to return an error only when the id is an empty string, you probably better do just checking for this in here then?